plugins/perceptionexpander.koplugin/main.lua

```diff
@@ -0,0 +1,232 @@
+local Widget = require("ui/widget/widget")
+local LineWidget = require("ui/widget/linewidget")
+local MultiInputDialog = require("ui/widget/multiinputdialog")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local HorizontalGroup = require("ui/widget/horizontalgroup")
+local UIManager = require("ui/uimanager")
+local InfoMessage = require("ui/widget/infomessage")
+local Geom = require("ui/geometry")
+local Screen = require("device").screen
+local T = require("ffi/util").template
+local _ = require("gettext")
+local Blitbuffer = require("ffi/blitbuffer")
+
+local PerceptionExpander = Widget:extend{
+    is_enabled = nil,
+    name = "percepton_expander",
+    page_counter = 0,
+    shift_each_pages = 100,
+    margin = 0.1,
+    line_thick = 2,
+    line_color_intensity = 0.3,
+    margin_shift = 0.03,
+}
+
+function PerceptionExpander:init()
+    local settings = G_reader_settings:readSetting("perception_expander") or {}
```

Comment:
I suggest rolling out perception expander's own setting file like what we do with other components: https://github.com/koreader/koreader/blob/4f64ac4b69790b93af5a2e3c82a17ae11eae0177/frontend/apps/cloudstorage/cloudstorage.lua#L351
https://github.com/koreader/koreader/blob/a13f55cfe8783ce42069f9c17d41f447af810751/frontend/ui/network/manager.lua#L14